### PR TITLE
feat(Request): add static factory method create to Request class

### LINKED ISSUE #5621: Http: add factory static methods to Request class
rather than:
```typescript
var request = new Request(new RequestArgs(etc));
```
enable:
```typescript
var request1 = Request.create(url, options)
```

### PR DESCRIPTION
**Pull request by vepasto**
Resolves #5621

### CODE REVIEW

File: modules/angular2/src/http/static_request.ts [hunk 1]
```diff
@@ -50,6 +50,15 @@ import {
  */
 export class Request {
   /**
+   * Static factory method
+   */
+  static create: Function =
```
Comment (PascalPrecht) [inline modules/angular2/src/http/static_request.ts]: Do we use `Function` type for function definitions?

**Comment by robwormald** [comment]
@vepasto thanks for the PR! I'm in the middle of refactoring the guts of Http so this might take a while to land.

This should, I think, accept either `(url:string, requestOptionsArgs:RequestOptionsArgs)` or simply `(requestOptionsArgs:RequestOptionsArgs)`, similar to how the native `fetch` implementation works: https://developer.mozilla.org/en-US/docs/Web/API/Request/Request

Thinking further, the constructor itself should probably accept the same. Feel like tackling that too?

**Comment by vepasto** [comment]
@robwormald I agree that it should be like the native one. I'll add overloading method.

That's a good idea to constructor accept the same. Should I create new issue for this or just use the same pull request?

Anyways I'll fix both of those.

**Comment by vepasto** [comment]
I realised that it is better to add overloading methods to constructor and then just call those in create method.

**Comment by tbosch** [comment]
@robwormald ping?

**Comment by mhevery** [comment]
@robwormald what is the status of this PR? Can we merge/rebase/close it?

**Pull request closed without merging**